mergify_engine/actions/merge_base.py

```diff
@@ -49,6 +49,7 @@
 MARKDOWN_TITLE_RE = re.compile(r"^#+ ", re.I)
 MARKDOWN_COMMIT_MESSAGE_RE = re.compile(r"^#+ Commit Message ?:?\s*$", re.I)
 REQUIRED_STATUS_RE = re.compile(r'Required status check "([^"]*)" is expected.')
+FORBIDDEN_MERGE_COMMITS_MSG = "Merge commits are not allowed on this repository."
```

Comment:
I'm pretty sure we should also handle `squash` or `rebase` not being allowed.

Reply:
I will add them.